src/ML-POD/compute_podfit.cpp

```diff
@@ -0,0 +1,1564 @@
+    if (keywd == "error_analysis_for_training_data_set") fitting_weights[3] = utils::numeric(FLERR,words[1],false,lmp);
+    if (keywd == "error_analysis_for_test_data_set") fitting_weights[4] = utils::numeric(FLERR,words[1],false,lmp);
+    if (keywd == "energy_force_calculation_for_training_data_set") fitting_weights[5] = utils::numeric(FLERR,words[1],false,lmp);
+    if (keywd == "energy_force_calculation_for_test_data_set") fitting_weights[6] = utils::numeric(FLERR,words[1],false,lmp);
+    if (keywd == "percentage_training_data_set") fitting_weights[7] = utils::numeric(FLERR,words[1],false,lmp);
+    if (keywd == "percentage_test_data_set") fitting_weights[8] = utils::numeric(FLERR,words[1],false,lmp);
+    if (keywd == "randomize_training_data_set") fitting_weights[9] = utils::numeric(FLERR,words[1],false,lmp);
+    if (keywd == "randomize_test_data_set") fitting_weights[10] = utils::numeric(FLERR,words[1],false,lmp);
+
+    // other settings
+
+    if (keywd == "file_format") file_format = words[1];
+    if (keywd == "file_extension") file_extension = words[1];
+    if (keywd == "path_to_training_data_set"){
+      utils::logmesg(lmp, "{}\n", words[1]);
+      training_path = words[1];
+      for (int i=2; i < words.size(); i++){
+        training_path = training_path + " " + words[i];
+      }
+    }
+    if (keywd == "path_to_test_data_set"){
+      test_path = words[1];
+      for (int i=2; i < words.size(); i++){
+        test_path = test_path + " " + words[i];
+      }
+    }
+
+  }
+
+  utils::logmesg(lmp, "**************** Begin of Data File ****************\n");
+  utils::logmesg(lmp, "file format: {}\n", file_format);
+  utils::logmesg(lmp, "file extension: {}\n", file_extension);
+  utils::logmesg(lmp, "path to training data set: {}\n", training_path);
+  utils::logmesg(lmp, "path to test data set: {}\n", test_path);
+  utils::logmesg(lmp, "training percentage: {}\n", fitting_weights[7]);
+  utils::logmesg(lmp, "test percentage: {}\n", fitting_weights[8]);
+  utils::logmesg(lmp, "randomize training data set: {}\n", fitting_weights[9]);
+  utils::logmesg(lmp, "randomize test data set: {}\n", fitting_weights[10]);
+  utils::logmesg(lmp, "error analysis for training data set: {}\n", fitting_weights[3]);
+  utils::logmesg(lmp, "error analysis for test data set: {}\n", fitting_weights[4]);
+  utils::logmesg(lmp, "energy/force calculation for training data set: {}\n", fitting_weights[5]);
+  utils::logmesg(lmp, "energy/force calculation for test data set: {}\n", fitting_weights[6]);
+  utils::logmesg(lmp, "fitting weight for energy: {}\n", fitting_weights[0]);
+  utils::logmesg(lmp, "fitting weight for force: {}\n", fitting_weights[1]);
+  utils::logmesg(lmp, "fitting weight for stress: {}\n", fitting_weights[2]);
+  utils::logmesg(lmp, "**************** End of Data File ****************\n");
+}
+
+std::vector<std::string> CPODFIT::globVector(const std::string& pattern, std::vector<std::string> & files)
```

Comment:
Convenience functions like this should not be class members. If there is only one application, they can be defined as static functions at the top of the file.  Also, the same functionality can be achieved with using "platform::list_directory()" and looping over the resulting vector of strings and processing only files that would match a "regex" pattern with "utils::strmatch()". That would be better code reuse and is platform neutral.

Reply:
It is defined as a static function at the top of the file.